File: Kyou44.cpp
#include<iostream>
#include<algorithm>
using namespace std;
int main(){
    int n = 0,m = 0,cnt = 0, b = 0, c = 0;
    long long ans = 0;
    cin >> n >> m;
    int a[n] = {0};

    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    sort(a, a+n);

    for (int i = 0; i < m; i++) {
        cnt = 0;
        cin >> b >> c;
        for (int j = 0; j < n; j++) {
            if (a[j] < c) {
                a[j] = c;
                cnt++;
                if (cnt >= b){
                    break;
                }
            }else{
                break;
            }

        }

        sort(a, a+n);
    }

    for (int i = 0; i < n; i++) {
        ans += a[i];
    }
    cout << ans << endl;



}

File: Kyou44.py
n,m = map(int, input().split())
a = list(map(int,input().split()))
a.sort()
ans = 0
tmp = []
tm = 0
for i in range(m):
    cnt = 0
    b,c = map(int, input().split())
    tmp.append([b,c])

tmp = sorted(tmp, reverse = True, key = lambda x : x[1])
cnt = 0
for i in range(n):
    if a[i] < tmp[cnt][1]:
        a[i] = tmp[cnt][1]
        tmp[cnt][0] -= 1
        if tmp[cnt][0] == 0:
            cnt += 1
            if cnt == m:
                break
    else:
        break

print(sum(a))
